contractcourt/commit_sweep_resolver.go

```diff
@@ -76,155 +139,109 @@ func (c *commitSweepResolver) Resolve() (ContractResolver, error) {
 		return nil, nil
 	}
 
-	// First, we'll register for a notification once the commitment output
-	// itself has been confirmed.
-	//
-	// TODO(roasbeef): instead sweep asap if remote commit? yeh
-	commitTXID := c.commitResolution.SelfOutPoint.Hash
-	sweepScript := c.commitResolution.SelfOutputSignDesc.Output.PkScript
-	confNtfn, err := c.Notifier.RegisterConfirmationsNtfn(
-		&commitTXID, sweepScript, 1, c.broadcastHeight,
-	)
+	confHeight, err := c.getCommitTxConfHeight()
 	if err != nil {
 		return nil, err
 	}
 
-	c.log.Debugf("waiting for commit tx to confirm")
+	unlockHeight := confHeight + c.commitResolution.MaturityDelay
 
-	select {
-	case _, ok := <-confNtfn.Confirmed:
-		if !ok {
-			return nil, errResolverShuttingDown
-		}
+	c.log.Debugf("commit conf_height=%v, unlock_height=%v",
+		confHeight, unlockHeight)
 
-	case <-c.quit:
-		return nil, errResolverShuttingDown
-	}
+	amt := btcutil.Amount(
+		c.commitResolution.SelfOutputSignDesc.Output.Value,
+	)
 
-	// We're dealing with our commitment transaction if the delay on the
-	// resolution isn't zero.
-	isLocalCommitTx := c.commitResolution.MaturityDelay != 0
+	c.reportLock.Lock()
+	c.currentReport = &ContractReport{
+		Outpoint:         c.commitResolution.SelfOutPoint,
```

Comment:
most of these values are available at constructor time, we can update only what is known first here. (perhaps with a method that acquired the mutex?)

Reply:
report initialized with what we know